site/client/src/main/scala/Model.scala

```diff
@@ -0,0 +1,32 @@
+package model
+
+case class ClientExercise(
+    method:    String,
+    arguments: Seq[String] = Nil
+) {
+
+  def isFilled: Boolean = !arguments.exists(_.isEmpty) && arguments.nonEmpty
+
+  def isSolved: Boolean = false
+
+  def canBeCompiled: Boolean = isFilled && !isSolved
+}
+
+class ExerciseState extends Enumeration {
```

Comment:
We should try to avoid using enumerations where possible as they may be the source of many issues down the road.
A better type for this case would be using an ADT withc ase objects.
```scala
sealed trait ExerciseState
case object Unsolved extends ExerciseState
...
```